### DISCUSSION TITLE
How do I delete phantomcamera host for next scene?

### DISCUSSION THREAD

**Discussion by mak448a** [ramokz/phantom-camera#423] [Question / Help Wanted] [answered]
After I called queue_free() on a sub-scene, I fade in a new level.
Even after doing that, it still gives me this error:
`Only one PhantomCameraHost can exist in a scene`

Please help!

**Comment by ramokz** [comment]
Would you be able to share either a small sample project, ideally with as little in it as possible, where that condition occurs?

**Reply by mak448a** [reply]
Sorry for not getting back to you sooner! If I don't forget, I'll try to make a project later.

**Reply by mak448a** [reply]
Attached is my project. I'm probably doing something wrong, but please help!
[mrp.zip](https://github.com/user-attachments/files/18031612/mrp.zip)

**Reply by ramokz** [reply, marked as answer]
The issue you're encountering is due to freeing `Scene1` using `queue_free()` and instantiating `Scene2` within the same method.
Basically, `queue_free()` only removes that node from the scene after the process has finished its tick/frame, meaning you're instantiating `Scene2` before `Scene1` has been freed.

One way would be to use `free()` instead as that will not wait until the end of the tick free the node, but generally `queue_free()` is safer to use.

So an alternative solution would be to just wait until `Scene1` has been removed before instantiating `Scene2`. Have added some sample code below that does that; in addition to simplifying the script a bit:

```gdscript
# Input should generally be done using either the _unhandled_input() or _input() function
# _process is a lot more computantially expensive and can easily introduce input bugs
func _input(event: InputEvent) -> void:

  if Input.is_action_just_pressed("ui_accept"):

    # No need to reference `get_tree().get_root().get_node("/root/Main")`
    # as this script is that refernece
    var scene_1 : Node2D = get_node("Scene1")

    # Uses queue_free() like before
    scene_1.queue_free()

    # Pauses the code here until Scene1 has exited from the SceneTree
    await scene_1.tree_exited

    # Loads the scene like before
    var hi: PackedScene = load("res://scene2.tscn")
		
    # Instantiates the scene like before
    add_child(hi.instantiate())
```
